src/biokbase/narrative/jobs/job.py

```diff
@@ -369,3 +436,82 @@ def dump(self):
         """
 
         return {attr: getattr(self, attr) for attr in [*ALL_JOB_ATTRS, "_last_state"]}
+
+    # def _create_error_state(
+    #     self,
+    #     error: str,
+    #     error_msg: str,
+    #     code: int,
+    # ) -> dict:
+    #     """
+    #     Creates an error state to return if
+    #     1. the state is missing or unretrievable
+    #     2. Job is none
+    #     This creates the whole state dictionary to return, as described in
+    #     _construct_cache_job_state.
+    #     :param error: the full, detailed error (not necessarily human-readable, maybe a stacktrace)
+    #     :param error_msg: a shortened error string, meant to be human-readable
+    #     :param code: int, an error code
+    #     """
+    #     return {
+    #         "status": "error",
+    #         "error": {
+    #             "code": code,
+    #             "name": "Job Error",
+    #             "message": error_msg,
+    #             "error": error,
+    #         },
+    #         "errormsg": error_msg,
+    #         "error_code": code,
+    #         "job_id": self.job_id,
+    #         "cell_id": self.cell_id,
+    #         "run_id": self.run_id,
+    #         "created": 0,
+    #         "updated": 0,
+    #     }
+
+    # def _child_states(self, sub_job_list):
+    #     """
+    #     Fetches state for all jobs in the list. These are expected to be child jobs, with no actual Job object associated.
+    #     So if they're done, we need to do the output mapping out of band.
+    #     But the check_jobs call with params will return the app id. So that helps.
+    #     app_id = the id of the app that all the child jobs are running (format: module/method, like "MEGAHIT/run_megahit")
+    #     app_tag = one of "release", "beta", "dev"
+    #     (the above two aren't stored with the subjob metadata, and won't until we back some more on KBParallel - I want to
+    #     lobby for pushing toward just starting everything up at once from here and letting HTCondor deal with allocation)
+    #     sub_job_list = list of ids of jobs to look up
+    #     """
+
+    #     # TODO
+    #     # THIS IS A DRAFT METHOD
+    #     # NEEDS TO BE RUN/VERIFIED
```

Comment:
similarly -- what's happening with this method? are you planning to reinstate it?

Reply:
I just corrected some bugs by eye and commented it out. I didn't know what to do with it. I can delete it :sweat_smile:

Reply:
I think we'll need some variant of it in the job manager to ensure that all the child jobs of a batch parent are in the running jobs list, but it's not worth keeping this code as-is. Delete!

Reply:
Yeah, what @ialarmedalien said. This can get fixed later in the job manager.